### PR TITLE
DP-1079 - Create a new schema example: UseSchemaFilterToValidateRecordsInAPipeline

### PR DESCRIPTION
**Pull request by gvjoshi25**

### CODE REVIEW

File: src/main/java/com/northconcepts/datapipeline/foundations/examples/schema/UseSchemaFilterToValidateRecordsInAPipeline.java [hunk 1]
```diff
@@ -0,0 +1,67 @@
...
+                .add(new FieldFilter("Variant Price")
+                    .addRule(new IsNotNull())
+                    .addRule(new IsJavaType(Double.class)))
+                
+                .add(new FilterExpression("${Variant Price} >= 5 && ${Variant Price} <= 50"))
+        
+                .add(new FieldFilter("Option1 Value")
+                        .addRule(new IsJavaType(String.class))
+                        .addRule(new ValueMatch<String>("Blue", "Black", "Gold", "Silver", "Purple")))
+
```
Comment (DeleTaylor) [inline src/main/java/com/northconcepts/datapipeline/foundations/examples/schema/UseSchemaFilterToValidateRecordsInAPipeline.java]: The non-schema validations are hiding the purpose of this example.  Are they needed?

File: src/main/java/com/northconcepts/datapipeline/foundations/examples/schema/UseSchemaFilterToValidateRecordsInAPipeline.java [hunk 2]
```diff
@@ -0,0 +1,67 @@
...
+        DataReader reader = new CSVReader(new File("data/input/jewelry.csv"))
+                .setAllowMultiLineText(true)
+                .setFieldNamesInFirstRow(true);
+        
+        reader = new TransformingReader(reader)
+                .add(new BasicFieldTransformer("Variant Price").stringToDouble(),
+                    new BasicFieldTransformer("Variant Taxable").stringToBoolean())
+                .add(new SelectFields("Variant Price", "Variant Taxable", "Title", "Option1 Value"));
+        
+        reader = new FilteringReader(reader)
```
Comment (DeleTaylor) [inline src/main/java/com/northconcepts/datapipeline/foundations/examples/schema/UseSchemaFilterToValidateRecordsInAPipeline.java]: Let's use a `ValidatingReader` instead.

**Review by DeleTaylor — CHANGES_REQUESTED** [review]

File: src/main/java/com/northconcepts/datapipeline/foundations/examples/schema/UseSchemaFilterToValidateRecordsInAPipeline.java (hunk 1 above)
Comment (gvjoshi25) [reply src/main/java/com/northconcepts/datapipeline/foundations/examples/schema/UseSchemaFilterToValidateRecordsInAPipeline.java]: Okay, I understand. I thought we have to prepare an example with such validation also... I will remove them.

File: src/main/java/com/northconcepts/datapipeline/foundations/examples/schema/UseSchemaFilterToValidateRecordsInAPipeline.java (hunk 2 above)
Comment (gvjoshi25) [reply src/main/java/com/northconcepts/datapipeline/foundations/examples/schema/UseSchemaFilterToValidateRecordsInAPipeline.java]: Done.

**Review by DeleTaylor — APPROVED** [review]

**Pull request merged**